### DISCUSSION TITLE
How to fully update a single topic?

### DISCUSSION THREAD

**Discussion by erikmd** [magit/forge#391] [Support] [answered]
I've just installed forge and after doing the initial pull with `f y`,
a given Pull Request has been updated (some commits were pushed outside of Magit to the GItHub remote),
but in this case, what forge command is recommended to ensure the latest version of the remote commits are fetched?

In particular, it seems `M-x forge-pull-topic` doesn't update the `pullreqs` ref;
while `f y` does trigger this update… albeit it is maybe overkill to run it if one only want to work on a single topic?

Thanks in advance for your reply and any related tip :)

**Comment by tarsius** [comment, marked as answer]
I do recommend `forge-pull` (`f y` or `N f f`) even if just one topic changed. `forge-pull-topic` (`N f t`) is for when you know that something changed, yet the API refuses to acknowledge that and keeps claiming that no topic changed, or in huge active repositories, where you want to avoid fetching other topics and git data (i.e. the pullreqs refs) because the keep changing constantly.

`forge-pull` may be a bit wasteful, but it happens in the background so that shouldn't be much of an issue.

**Reply by erikmd** [reply]
OK, thanks for your advice!

But just for completeness, is there a magit or forge shortcut (or maybe just a `git fetch` command :) we could run to only fetch one or all (I'm OK with both) `pullreq` refs?

**Reply by tarsius** [reply]
Yes. Most of them. Fetching the pull-request refs works by adding `+refs/pull/*/head:refs/pullreqs/*` to `remote.origin.fetch` (or whatever the upstream remote is named). So fetching from the upstream also fetches these refs. I almost aalways use `f a`, but `f u` would work too.
